includes/express-checkout/class-wc-payments-express-checkout-ajax-handler.php

```diff
@@ -48,26 +48,38 @@ public function init() {
 
 	/**
 	 * Create order. Security is handled by WC.
+	 *
+	 * @throws Exception If cart is empty.
 	 */
 	public function ajax_create_order() {
-		if ( WC()->cart->is_empty() ) {
-			wp_send_json_error( __( 'Empty cart', 'woocommerce-payments' ), 400 );
-		}
+		try {
+			if ( WC()->cart->is_empty() ) {
+				throw new Exception( __( 'Empty cart', 'woocommerce-payments' ) );
+			}
 
-		if ( ! defined( 'WOOCOMMERCE_CHECKOUT' ) ) {
-			define( 'WOOCOMMERCE_CHECKOUT', true );
-		}
+			if ( ! defined( 'WOOCOMMERCE_CHECKOUT' ) ) {
+				define( 'WOOCOMMERCE_CHECKOUT', true );
+			}
 
-		if ( ! defined( 'WCPAY_ECE_CHECKOUT' ) ) {
-			define( 'WCPAY_ECE_CHECKOUT', true );
-		}
+			if ( ! defined( 'WCPAY_ECE_CHECKOUT' ) ) {
+				define( 'WCPAY_ECE_CHECKOUT', true );
+			}
 
-		// In case the state is required, but is missing, add a more descriptive error notice.
-		$this->express_checkout_button_helper->validate_state();
+			// In case the state is required, but is missing, add a more descriptive error notice.
+			$this->express_checkout_button_helper->validate_state();
 
-		$this->express_checkout_button_helper->normalize_state();
+			$this->express_checkout_button_helper->normalize_state();
 
-		WC()->checkout()->process_checkout();
+			WC()->checkout()->process_checkout();
+		} catch ( Exception $e ) {
+			Logger::error( 'Failed to process express checkout payment: ' . $e );
+
+			$response = [
+				'result'   => 'error',
+				'messages' => $e->getMessage(),
+			];
+			wp_send_json( $response, 400 );
```

Comment:
Just noting that this is the [response expected](https://github.com/Automattic/woocommerce-payments/blob/6d4e0544cf643255fa53f04a8255251d8c489506/client/express-checkout/event-handlers.js#L95-L100) in the frontend.